internal/backend/backend_retry.go

```diff
@@ -128,3 +128,17 @@ func (be *RetryBackend) Test(ctx context.Context, h restic.Handle) (exists bool,
 	})
 	return exists, err
 }
+
+// List runs fn for each file in the backend which has the type t.
+func (be *RetryBackend) List(ctx context.Context, t restic.FileType, fn func(restic.FileInfo) error) error {
+	listed := make(map[string]bool)
```

Comment:
We can improve this: The `bool` is not needed, just use an empty struct:
```go
listed := make(map[string]struct{})
```

and below:
```
listed[fi.Name] = struct{}{}
```

Reply:
fixed.